hw/chip/mimxrt10xx/family.c

```diff
@@ -424,22 +443,14 @@ void board_check_app_start(void)
 #ifdef BOARD_LED_ON_UF2_START
     board_led_write(true);
 #endif
+
     board_delay_ms(BOARD_TAP_WAIT);
+
 #ifdef BOARD_LED_ON_UF2_START
     board_led_write(false);
 #endif
-  }
 
-
-  // Boot into the application
-  _bootloader_dbl_tap = 0;
-
-  /* Rebase the Stack Pointer */
-  __set_MSP(*(uint32_t *)APP_START_ADDRESS);
-
-  /* Rebase the vector table base address */
-  SCB->VTOR = ((uint32_t)APP_START_ADDRESS & SCB_VTOR_TBLOFF_Msk);
-
-  /* Jump to application Reset Handler in the application */
-  asm("bx %0" ::"r"(app_start_address));
+  // If we made to to here then we should boot into the application
+  _bootloader_dbl_tap = DBL_TAP_MAGIC_QUICK_BOOT;
+  board_reset();
```

Comment:
So we reset twice, I guess that's ok.

Reply:
Yep, added a note to that effect. Means the platform comes up 'clean' without us having messed with it.